server/service/orbit.go

```diff
@@ -181,6 +182,33 @@ func (svc *Service) GetOrbitConfig(ctx context.Context) (fleet.OrbitConfig, erro
 		return fleet.OrbitConfig{}, err
 	}
 
+	var hostOS *fleet.OperatingSystem
+	if host.ID != 0 {
```

Comment:
I believe we can assume you'll have a valid host struct at this point.

The way this works is via this middleware:

https://github.com/fleetdm/fleet/blob/f59cc061513279ff92453589faa11f6e68e8cbc8/server/service/handler.go#L787-L788

which stores the host in the context here:

https://github.com/fleetdm/fleet/blob/f59cc061513279ff92453589faa11f6e68e8cbc8/server/service/endpoint_middleware.go#L132-L147

The datastore method that loads the host eventually is:

https://github.com/fleetdm/fleet/blob/f59cc061513279ff92453589faa11f6e68e8cbc8/server/datastore/mysql/hosts.go#L2096-L2101

I think it would be desirable to fail here if any of the steps above failed to catch an error.

Reply:
I originally didn't have that guard clause, but added it because for some reason certain integration tests didn't mock the operating systems table correctly and it would cause a null pointer dereference during test runs. I can remove it if I go around and fix them though